lib/src/rules/use_build_context_synchronously.dart

```diff
@@ -38,6 +40,20 @@ void onButtonTapped(BuildContext context) {
 }
 ```
 
+**GOOD:**
+```dart
+void onButtonTapped() async {
+  final navigator = Navigator.of(context);
+  await Future.delayed(const Duration(seconds: 1));
+
+  navigator.pop();
+}
+```
```

Comment:
This is NOT good practice. The navigator could now become invalid during the async gap (the `await`) and your code crashes. You should not hold on to the navigator like that and instead look it up after the async gap and after checking that the context is still valid.